tests/Google/Task/ComposerTest.php

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class Google_Task_ComposerTest extends BaseTest
+{
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage Google service "Foo" does not exist
+     */
+    public function testInvalidServiceName()
+    {
+        Google_Task_Composer::cleanup($this->createMockEvent(['Foo']));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage Invalid Google service name "../YouTube"
+     */
+    public function testRelatePathServiceName()
+    {
+        Google_Task_Composer::cleanup($this->createMockEvent(['../YouTube']));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage Google service "" does not exist
+     */
+    public function testEmptyServiceName()
+    {
+        Google_Task_Composer::cleanup($this->createMockEvent(['']));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage Invalid Google service name "YouTube*"
+     */
+    public function testWildcardServiceName()
+    {
+        Google_Task_Composer::cleanup($this->createMockEvent(['YouTube*']));
+    }
+
+    public function testRemoveServices()
+    {
+        $vendorDir = sys_get_temp_dir() . '/rand-' . rand();
+        $serviceDir = sprintf(
+            '%s/google/apiclient-services/src/Google/Service/',
+            $vendorDir
+        );
+        $dirs = [
+            'ServiceToKeep',
+            'ServiceToDelete1',
+            'ServiceToDelete2',
+        ];
+        $files = [
+            'ServiceToKeep/ServiceFoo.php',
+            'ServiceToKeep.php',
+            'SomeRandomFile.txt',
+            'ServiceToDelete1/ServiceFoo.php',
+            'ServiceToDelete1.php',
+            'ServiceToDelete2/ServiceFoo.php',
+            'ServiceToDelete2.php',
+        ];
+        foreach ($dirs as $dir) {
+            @mkdir($serviceDir . $dir, 0777, true);
+        }
+        foreach ($files as $file) {
+            touch($serviceDir . $file);
+        }
+        $print = 'Removing 2 google services';
+        Google_Task_Composer::cleanup(
+            $this->createMockEvent(['ServiceToKeep'], $vendorDir, $print),
+            $this->createMockFilesystem([
+                'ServiceToDelete2',
+                'ServiceToDelete2.php',
+                'ServiceToDelete1',
+                'ServiceToDelete1.php',
+            ], $serviceDir)
+        );
+    }
+
+    private function createMockFilesystem(array $files, $serviceDir)
+    {
+        $consecutive = [];
+        foreach ($files as $filename) {
+            $file = new \SplFileInfo($serviceDir . $filename);
+            $consecutive[] = [$this->equalTo($file->getRealPath())];
+        }
+        $mockFS = $this->getMockBuilder('Symfony\Component\Filesystem\Filesystem')
```

Comment:
PHPUnit's Mock Builder isn't really compatible with PHP 7.4. You should use Prophecy. The rest of the unit tests have been migrated already.

Reply:
Is Prophecy compatible all the way back to 5.4?

Reply:
Yep